src/sentry/api/endpoints/project_repo_path_parsing.py

```diff
@@ -11,41 +11,10 @@
 from sentry.api.bases.project import ProjectEndpoint, ProjectPermission
 from sentry.api.serializers.rest_framework.base import CamelSnakeSerializer
 from sentry.integrations import IntegrationFeatures
+from sentry.integrations.utils.code_mapping import find_roots
 from sentry.models.repository import Repository
 from sentry.services.hybrid_cloud.integration import RpcIntegration, integration_service
 
-SLASH = "/"
-BACKSLASH = "\\"  # This is the Python representation of a single backslash
-
-
-def find_roots(stack_path, source_path):
-    """
-    Returns a tuple containing the stack_root, and the source_root.
-    If there is no overlap, raise an exception since this should not happen
-    """
-    stack_path_delim = SLASH if SLASH in stack_path else BACKSLASH
-    overlap_to_check = stack_path.split(stack_path_delim)
-    stack_root = []
-    while overlap_to_check:
-        if source_path.endswith(overlap := SLASH.join(overlap_to_check)):
-            source_root = source_path.rpartition(overlap)[0]
-            stack_root = stack_path_delim.join(stack_root)
-            if not source_root:  # replace empty source root with "slash"
-                source_root = SLASH
-            if not stack_root:  # replace empty stack root with "dot slash"
-                stack_root = f".{stack_path_delim}"
```

Comment:
These two lines were removed to reflect the fact that source_root and stack_root should be empty when roots match

Reply:
What I tried mentioning in the call (I possibly did not say it or I explained incorrectly) is to have first a PR for the fix so it's easy to see what gets changed and a follow-up PR for any refactoring. This makes it easier for the reviewer to see what changes.

Don't worry about it. I will review it as is

Reply:
For anyone coming along, this changes the addition of these lines:
https://github.com/getsentry/sentry/pull/65474/files#diff-08924ad13c52b755328b00053d214294721782bac7af4600e3f3ad4599b3180bR33-R38

Reply:
Ah I see what you mean now, I realize might've jumped the gun on a couple of these changes and they could've waited for a future PR... sorry about that!